File: 88-merge-sorted-array/88-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int a=0;
        int b=0;
        int c=m;
         if(nums2.length==0){
            return;
        }
        else if(m==0){
            for(int i=0;i<n;i++){
                nums1[i]=nums2[i];
            }
        }
        else{
        while(a<nums1.length && b<n){
            if(nums1[a]>nums2[b] || (nums2[b]==0 && nums1[a]==nums2[b]) || (nums1[a]==0 && a>=m)){
                nums1[m]=nums1[m-1];
                for(int i=m-1;i>a;i--){
                    nums1[i]=nums1[i-1];
                }
                nums1[a]=nums2[b];
                a++;
                b++;
                m++;
            }
            else if(nums1[a]<nums2[b]){
                a++;
}
            else{
                a++;
            }
        }
        }
    }
}

File: 88-merge-sorted-array/88-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        vector<int> result;
        int start1 =0;
        int start2 = 0;
        int end1=m-1;
        int end2=n-1;
        while(start1<=end1 && start2<=end2){
            if(nums1[start1]<nums2[start2]){
                result.push_back(nums1[start1]);
                start1++;
            }
            else{
                result.push_back(nums2[start2]);
                start2++;
            }
            
        }
        while(start1<=end1){
            result.push_back(nums1[start1]);
                start1++;
        }
        while(start2<=end2){
            result.push_back(nums2[start2]);
                start2++;
        }
        for(int k=0;k<result.size();k++){
            nums1[k] = result[k];
        }
        
    }
};